string_test.go

```diff
@@ -158,17 +159,29 @@ func TestStringEncode(t *testing.T) {
 	testJavaDecode(t, "argString_65536", s65560[:65536])
 }
 
-func BenchmarkDecodeStringOptimized(t *testing.B) {
+var decodePool = &sync.Pool{
```

Comment:
the pool seems only being used once, is it necessary?

Reply:
This is used to test pooling decoding and improve code coverage